### ISSUE TITLE
Deep Learning 

### ISSUE THREAD

**Issue by HugoRodriguesQW** [HugoRodriguesQW/neuron#2]
<p align='center'>
<img  align='top' src='https://github.com/HugoRodriguesQW/neuron/assets/71078903/386c03f4-36f3-44f9-b7ec-9bd177eaebb4'></img>
</p>

### Natural Selecion (built-in package)

1 - Generate multiple AIs simultaneously by doing their interactions:
Yes, in the genetic algorithm approach to tuning hyperparameters, you can create multiple instances of AIs (neural networks) with different hyperparameter settings. Each of these is called an "individual" or "chromosome".

2 - The closer to your goal or when you reach it, it increases your fitness:
Each individual (IA) is evaluated based on their performance on a specific metric (fitness), which can be classification accuracy, average error, accuracy or any other metric relevant to the problem at hand. The closer to the goal or the better the performance on the fitness metric, the higher the fitness value assigned to the individual.

3 - They will be divided into generations, which can be passed manually or automatically:
Yes, after evaluating the performance of each individual, you can create a new generation of AIs through processes of selection, crossover (genetic recombination) and mutation. This generation will be composed of AIs who inherit traits from the most successful individuals of the previous generation. This process can be done manually or automatically, following the rules and parameters defined in the genetic algorithm.

4 - The strongest (%) or strongest becomes the new default:
Generally, you select the strongest individuals (those with the highest fitness value) to make up the next generation. It can be a selection by percentage (the best X%) or just the individual with the best fitness. These individuals will contribute their genes (network weights) to the next generation, helping to improve overall performance over generations.

5 - The weakest remaining ones are eliminated:
Yes, in natural selection, the less fit individuals (with lower fitness value) can be eliminated, depending on the selection rules and survival strategies defined in the genetic algorithm. This helps ensure that only the most successful traits are passed down to the next generation, allowing performance to improve over time.

6 - This genetic algorithm optimization approach can be an efficient way to find good hyperparameter combinations for your neural networks, especially when the hyperparameter space is large and complex. However, keep in mind that implementing these algorithms requires care, tuning, and experimentation, as effectiveness can vary depending on the specific problem and the configurations used.
